packages/ra-ui-materialui/src/list/Pagination.spec.js

```diff
@@ -43,19 +43,50 @@ describe('<Pagination />', () => {
             expect(queryByText('ra.navigation.no_results')).toBeNull();
         });
 
-        it('should not display a pagination limit on an out of bounds page', () => {
+        it('should display a pagination limit on an out of bounds page (more than total pages)', async () => {
             jest.spyOn(console, 'error').mockImplementationOnce(() => {});
+            const setPage = jest.fn().mockReturnValue(null);
             const { queryByText } = render(
                 <ThemeProvider theme={theme}>
                     <ListContext.Provider
-                        value={{ ...defaultProps, total: 10, page: 2 }}
+                        value={{
+                            ...defaultProps,
+                            total: 10,
+                            page: 2, // Query the page 2 but there is only 1 page
+                            perPage: 10,
+                            setPage,
+                        }}
                     >
                         <Pagination />
                     </ListContext.Provider>
                 </ThemeProvider>
             );
-            // mui TablePagination displays a warning in that case, and that's normal
-            expect(queryByText('ra.navigation.no_results')).toBeNull();
+            // mui TablePagination displays no more a warning in that case
+            // Then useEffect fallbacks on a valid page
+            expect(queryByText('ra.navigation.no_results')).not.toBeNull();
```

Comment:
don't understand why we display the "no_result" if we fallback on a valid page 1